esphome/components/xiaomi_hhccjcy10/xiaomi_hhccjcy10.cpp

```diff
@@ -0,0 +1,60 @@
+#include "xiaomi_hhccjcy10.h"
+#include "esphome/core/log.h"
+#include "esphome/core/helpers.h"
+
+#ifdef USE_ESP32
+
+namespace esphome {
+namespace xiaomi_hhccjcy10 {
+
+static const char *const TAG = "xiaomi_hhccjcy10";
+
+void XiaomiHHCCJCY10::dump_config() {
+  ESP_LOGCONFIG(TAG, "Xiaomi HHCCJCY10");
+  LOG_SENSOR("  ", "Temperature", this->temperature_);
+  LOG_SENSOR("  ", "Moisture", this->moisture_);
+  LOG_SENSOR("  ", "Conductivity", this->conductivity_);
+  LOG_SENSOR("  ", "Illuminance", this->illuminance_);
+  LOG_SENSOR("  ", "Battery Level", this->battery_level_);
+}
+
+bool XiaomiHHCCJCY10::parse_device(const esp32_ble_tracker::ESPBTDevice &device) {
+  if (device.address_uint64() != this->address_) {
+    ESP_LOGVV(TAG, "parse_device(): unknown MAC address.");
+    return false;
+  }
+  ESP_LOGVV(TAG, "parse_device(): MAC address %s found.", device.address_str().c_str());
+
+  bool success = false;
+  for (auto &service_data : device.get_service_datas()) {
+    if (!service_data.uuid.contains(0x50, 0xFD)) {
+      ESP_LOGVV(TAG, "no tuya service data UUID.");
+      continue;
+    }
+    if (service_data.data.size() != 9) {  // tuya alternate between two service data
+      continue;
+    }
+    const uint8_t *data = service_data.data.data();
+
+    const int16_t temperature = encode_uint16(data[1], data[2]);
+    this->temperature_->publish_state((float) temperature / 10.0f);
```

Comment:
These need to be checked that the user actually configured them
```suggestion
    if (this->temperature_ != nullptr) {
      const int16_t temperature = encode_uint16(data[1], data[2]);
      this->temperature_->publish_state((float) temperature / 10.0f);
    }
```

Reply:
done